updated links in the PR template

toydarian opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issue-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
6. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:
Some links in the PR template are outdated.

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #5613 

**Special notes for your reviewer**:
I had to change a few links from git.k8s.io to github.com as the anchors would not work otherwise.

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
NONE
```

toydarian commented:
/assign @atoms

Atoms commented:
why not leave short links like: https://git.k8s.io/ ?

toydarian commented:
For some reason, the anchors (e.g. `#issue-kind-label`) do not seem to work with the short link sometimes. I land at the top of the page, not on the correct section.

Atoms commented:
that anchor is different now as it seems, i tested all new updated links with updated anchors and they work

toydarian commented:
My terminal emulator removes the parts behind the `#` sign, that's why I thought, the anchors do not work. I changed the links, squashed the commit and force-pushed it. I hope, that is not against the rules, but I want to keep the number of commits to a minimum.

Atoms commented:
/lgtm
/approve
All good now, thanks

LuckySB commented:
/approve

LuckySB commented:
/retest

LuckySB commented:
/ok-to-test

LuckySB commented:
/test packet_ubuntu16-canal-kubeadm
